app/src/main/java/com/nexters/zzallang/harusal2/ui/appwidget/TypeThreeWidgetProvider.kt

```diff
@@ -0,0 +1,103 @@
+package com.nexters.zzallang.harusal2.ui.appwidget
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.Context
+import android.content.Intent
+import android.widget.RemoteViews
+import com.nexters.zzallang.harusal2.R
+import com.nexters.zzallang.harusal2.application.util.NumberUtils
+import com.nexters.zzallang.harusal2.constant.SpendState
+import com.nexters.zzallang.harusal2.data.repository.BudgetRepository
+import com.nexters.zzallang.harusal2.data.repository.StatementRepository
+import com.nexters.zzallang.harusal2.ui.splash.SplashActivity
+import com.nexters.zzallang.harusal2.usecase.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class TypeThreeWidgetProvider : AppWidgetProvider() {
+	companion object {
+		const val THREE_REQUEST_CODE = 100
+	}
+
+	private val budgetRepository = BudgetRepository()
+	private val statementRepository = StatementRepository()
+
+	private val getTodayBudgetUseCase = GetTodayBudgetUseCase(budgetRepository, statementRepository)
+	private val getRemainMoneyUseCase = GetRemainMoneyUseCase(budgetRepository, statementRepository)
+	private val getSpentMoneyStatusUseCase = GetSpentMoneyStatusUseCase()
```

Comment:
단톡에서 따로 이야기했던거처럼 Koin을 이용해서 이렇게 해봅시당

```suggestion
class TypeThreeWidgetProvider(
    private val budgetRepository: BudgetRepository,
    private val statementRepository: StatementRepository,
    private val getTodayBudgetUseCase: GetTodayBudgetUseCase,
    private val getRemainMoneyUseCase: GetRemainMoneyUseCase,
    private val getSpentMoneyStatusUseCase: GetSpentMoneyStatusUseCase
) : AppWidgetProvider() {
	companion object {
		const val THREE_REQUEST_CODE = 100
	}
```